contracts/orderbook/Logic/OrderBookOrderPlace.sol

```diff
@@ -0,0 +1,258 @@
+pragma solidity ^0.8.0;
+import "../Storage/OrderBookEvents.sol";
+import "../Storage/OrderBookStorage.sol";
+import "../OrderVault/IDeposits.sol";
+
+contract OrderBookOrderPlace is OrderBookEvents, OrderBookStorage {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    function initialize(address target) public onlyOwner {
+        _setTarget(this.placeOrder.selector, target);
+        _setTarget(this.amendOrder.selector, target);
+        _setTarget(this.cancelOrder.selector, target);
+        _setTarget(this.cancelOrderProtocol.selector, target);
+        _setTarget(this.changeStopType.selector, target);
+        _setTarget(this.recoverFundsFromFailedOrder.selector, target);
+    }
+
+    function queryRateReturn(
+        address start,
+        address end,
+        uint256 amount
+    ) public view returns (uint256) {
+        (uint256 executionPrice, uint256 precision) = IPriceFeeds(
+            protocol.priceFeeds()
+        ).queryRate(start, end);
+        return (executionPrice * amount) / precision;
+    }
+
+    function _caseChecks(bytes32 ID, address collateral, address loanToken)
+        internal
+        view
+        returns (bool)
+    {
+        return 
+            protocol.getLoan(ID).loanToken ==
+            loanToken &&
```

Comment:
what are you checking exactly with loanToken && ? probably more readable !=address(0), not sure why would you check that